### PR TITLE
Docker Things

### LINKED ISSUE #209: Make package.json changes invalidate docker image

### PR DESCRIPTION
**Pull request by Jordan141**
fixes #209 

I think?

Adds scripts to package.json
- up: starts docker with --build param
- down: downs docker containers
- validate: checks package.json and compares it to see if there's been changes, tries rebuild if there has been

### CODE REVIEW

File: package.json
Comment (michaelKurowski) [inline package.json:10]: Could we not detach by default? Ctrl + P + Q is detaching, but to attach back it's a bit more cumbersome.

File: package.json
Comment (michaelKurowski) [inline package.json:12]: Why do we validate, did we not agree that --build solve all the problems?

File: devops/smart-install.sh
Comment (michaelKurowski) [inline devops/smart-install.sh:3]: This will work only if you're having the latest master fetched, and you're not on the master commit.

File: package.json
Comment (Jordan141) [reply package.json:12]: I was unsure if the behaviour met the criteria for the linked ticket. Do you want me to remove it?

Comment (michaelKurowski) [reply package.json:12]: I'd say so. You can remove validate and  smart-install, and consider ticket to be closed after mergint this PR. I'll approve it.

**Review by michaelKurowski — APPROVED** [review]

**Comment by michaelKurowski** [comment]
Approving in advance, please comply with the finding before merging.

**Pull request merged**
